packages/preview-middleware-client/types/sap.ui.dt.d.ts

```diff
@@ -1,16 +1,76 @@
 declare module 'sap/ui/dt/DesignTimeMetadata' {
     import type ManagedObject from 'sap/ui/base/ManagedObject';
-
+    import type AppComponent from 'sap/fe/core/AppComponent';
     export interface DesignTimeMetadataData {
         name: string;
         ignore: boolean;
         defaultValue: unknown;
         deprecated?: boolean;
         specialIndexHandling?: boolean;
     }
+    export enum TemplateType {
+        ListReport = 'ListReport',
+        ObjectPage = 'ObjectPage',
+        AnalyticalListPage = 'AnalyticalListPage',
+        FreeStylePage = 'None'
+    }
+    export interface DesigntimeSettingEnums {
+        id: string;
+        name: string;
+        description?: string;
+    };
+    export interface DesigntimeSetting {
```

Comment:
Are these the interfaces that Fiori Elements expose via design time metadata? If so don't they provide type definitions? It would be good to avoid having to redefine them here.

Reply:
As we saw yesterday. type definitions are not published for them